src/frequenz/client/electricity_trading/_client.py

```diff
@@ -339,6 +341,18 @@ def validate_params(  # pylint: disable=too-many-arguments
         if not isinstance(valid_until, _Sentinel) and valid_until is not None:
             if valid_until < datetime.now(timezone.utc):
                 raise ValueError("valid_until must be in the future")
+            if (
+                not isinstance(execution_option, _Sentinel)
+                and execution_option is not None
+            ):
+                if execution_option in [
+                    OrderExecutionOption.AON,
+                    OrderExecutionOption.FOK,
+                    OrderExecutionOption.IOC,
+                ]:
+                    raise ValueError(
+                        "valid_until must be None when execution_option is set to AON, FOK, or IOC"
+                    )
```

Comment:
I'd suggest to change the order of these checks. You might want to check first the execution option and after that  the valid until. Otherwise the current order of checks could lead to the user to try to set a valid `valid_until` while the execution option they are interesting in might not really need it.

Reply:
yea fair enough, this was to avoid repetition (as otherwise I need to check for `valid_until` being not None twice) but I don't have strong opinions about this

Reply:
This would be the code otherwise, you prefer this then?
```
        if (
            not isinstance(execution_option, _Sentinel)
            and execution_option is not None
            and execution_option
            in [
                OrderExecutionOption.AON,
                OrderExecutionOption.FOK,
                OrderExecutionOption.IOC,
            ]
        ):
            if not isinstance(valid_until, _Sentinel) and valid_until is not None:
                raise ValueError(
                    "valid_until must be None when execution_option is set to AON, FOK, or IOC"
                )
        if not isinstance(valid_until, _Sentinel) and valid_until is not None:
            if valid_until < datetime.now(timezone.utc):
                raise ValueError("valid_until must be in the future")
```

Reply:
ahhh I see, sorry for the confusion, will change it then!